Sources/ArcGISToolkit/FloorFilter/FloorFilterViewController.swift

```diff
@@ -0,0 +1,511 @@
+                    self.state = FloorFilterState.initiallyCollapsed
+                }
+            default:
+                break
+            }
+        }
+    }
+    
+    /// Static method that will be used to initialize the Floor Filter View and attach it as a SubView.
+    public static func makeFloorFilterView(
+        geoView: AGSGeoView?,
+        expansionDirection: ExpansionDirection = .up
+    ) -> FloorFilterViewController {
+        let storyboard = UIStoryboard(name: "FloorFilter", bundle: .module)
+        let floorFilterVC: FloorFilterViewController = storyboard.instantiateViewController(identifier: "FloorFilter")
+        floorFilterVC.expansionDirection = expansionDirection
+        floorFilterVC.geoView = geoView
+        
+        return floorFilterVC
+    }
+    
+    @available(*, unavailable)
+    override public init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        fatalError("use the method `makeFloorFilterView` instead")
+    }
+    
+    @available(*, unavailable)
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        // Update the views that are visibile and their heights based on the state of the Floor Filter.
+        updateViewsVisibilityForState(state: state)
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Initialize the UI elements and on click listeners.
+        initializeSiteButton()
+        initializeLevelsTableView()
+        initializeButtonsClickListeners()
+        buttonSizeDidChange()
+        
+        // Update the views that are visibile and their heights based on the state of the Floor Filter.
+        updateViewsVisibilityForState(state: state)
+        
+        // Adjust the constraints and order of the views in the Floor Filter if placement is on top or bottom of the sceen.
+        adjustConstraintsBasedOnPlacement()
+    }
+    
+    private func initializeFloorManager() {
+        guard let floorManager = floorManager else { return }
+        
+        floorManager.load(completion: { error in
+            if error != nil || floorManager.loadStatus != .loaded {
+                return
+            }
+            self.initializeSiteButton()
+                                        
+            // Filter the map to any previously selected level.
+            self.viewModel.filterMapToSelectedLevel()
+            self.levelsTableView.reloadData()
+        })
+    }
+    
+    private func initializeSiteButton() {
+        // Enable the site button if both the floor manager and the map is loaded.
+        if (viewModel.floorManager != nil && (geoView as? AGSMapView)?.map?.loadStatus == .loaded) {
+            addShadow()
+            siteBtn.backgroundColor = backgroundColor.withAlphaComponent(0.9)
+            siteBtn.isUserInteractionEnabled = true
```

Comment:
I just noticed this. Instead of disabling user interaction on the button, the button's `isEnabled` property should be set. That way, when the button is disabled, it looks like a disabled button. Just disabling user interaction doesn't change the button's appearance. Compare:

`isUserInteractionEnabled` | `isEnabled`
--- | ---
![Screen Shot 2022-02-04 at 8 19 35 AM](https://user-images.githubusercontent.com/2257493/152564468-9155fef0-5e3e-49ea-a730-66bca3f1bc1f.png) | ![Screen Shot 2022-02-04 at 8 15 14 AM](https://user-images.githubusercontent.com/2257493/152564496-a44a6c15-1cd1-4aca-8b92-c95b5a8c3bbc.png)